app/src/main/java/com/kks/nimblesurveyjetpackcompose/di/NetworkModule.kt

```diff
@@ -0,0 +1,159 @@
+import com.kks.nimblesurveyjetpackcompose.repo.token.TokenRepoImpl
+import com.kks.nimblesurveyjetpackcompose.util.CustomKeyProvider
+import com.kks.nimblesurveyjetpackcompose.util.CustomKeyProviderImpl
+import com.kks.nimblesurveyjetpackcompose.util.PreferenceManager
+import com.kks.nimblesurveyjetpackcompose.util.interceptors.AccessTokenInterceptor
+import com.kks.nimblesurveyjetpackcompose.util.interceptors.TokenAuthenticator
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import okhttp3.Authenticator
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+private const val TIMEOUT_SECONDS = 15L
+private const val CHUCKER_MAX_CONTENT_LENGTH = 250000L
+
+@Module
+@InstallIn(SingletonComponent::class)
+interface NetworkModule {
+
+    @Binds
+    fun provideTokenAuthenticator(tokenAuthenticator: TokenAuthenticator): Authenticator
+
+    @Binds
+    fun provideAuthAccessTokenInterceptor(accessTokenInterceptor: AccessTokenInterceptor): Interceptor
+
+    @Suppress("TooManyFunctions")
+    companion object {
+        @Singleton
+        @Provides
+        fun providePreferenceManager(@ApplicationContext appContext: Context): PreferenceManager =
+            PreferenceManager(appContext)
+
+        @ServiceQualifier
+        @Provides
+        fun provideServiceRetrofit(
+            @ServiceQualifier okHttpClient: OkHttpClient,
+            moshi: Moshi
+        ): Retrofit {
+            return Retrofit.Builder().baseUrl(BuildConfig.BASE_URL)
+                .client(okHttpClient)
+                .addConverterFactory(MoshiConverterFactory.create(moshi))
+                .build()
+        }
+
+        @AuthQualifier
+        @Provides
+        fun provideAuthRetrofit(
+            @AuthQualifier okHttpClient: OkHttpClient,
+            moshi: Moshi
+        ): Retrofit {
+            return Retrofit.Builder().baseUrl(BuildConfig.BASE_URL)
+                .client(okHttpClient)
+                .addConverterFactory(MoshiConverterFactory.create(moshi))
+                .build()
+        }
+
+        @Singleton
+        @Provides
+        fun providesMoshi(): Moshi {
+            return Moshi.Builder()
+                .addLast(KotlinJsonAdapterFactory())
+                .build()
+        }
+
+        @ServiceQualifier
+        @Provides
+        fun provideServiceOkHttpClient(
+            loggingInterceptor: HttpLoggingInterceptor,
+            chuckerInterceptor: ChuckerInterceptor,
+            tokenAuthenticator: TokenAuthenticator,
```

Comment:
```suggestion
            tokenAuthenticator: Authenticator,
```

Reply:
I am going to leave the naming like that to see the difference between the the from `Okhttp` and my custom one.

Reply:
I've changed in [Home Screen PR](https://github.com/kaungkhantsoe/NimbleSurveyJetpackCompose/pull/74)